### ISSUE TITLE
Certificate deployment seems to be broken on 4.8.0

### ISSUE THREAD

**Issue by Hedius** [wazuh/wazuh-puppet#1038]
Hello,

with 4.8.0 it seems like the cert deployment was heavily modified:

https://github.com/wazuh/wazuh-puppet/blob/0a3aa36c1779cd0e9afcfbbd0bb9445876e25109/manifests/certificates.pp#L37-L45

Especially this block.

Few things: 
* You hardcoded the environment in the repo. (Not everyone has a env called production)
* You seem to try to copy files from the puppet agent to the master? Eh you cannot do that? When you apply a catalog on a client you cannot magically place files on the master?
* It seems like the files are being tried to save in the files folder of another puppet module.?
* Would not be the correct way to share certs between nodes, etc. to use puppetdb and exported resources?
i can add more information if needed. But i am not sure how this is supposed to work if your wazuh host != the puppet master?

**Comment by Hedius** [comment]
https://documentation.wazuh.com/current/deployment-options/deploying-with-puppet/wazuh-puppet-module/index.html

Still exported resources would be cleaner. The setup does not work like this if you have a multi puppet master/server environment.

 please do not hijack other modules... and do not hardcode environments (puppet-archive in this case). :(

**Comment by pearsondavid** [comment]
`Indexer.pp` is in need of similar refactoring, it includes hardcoded puppet environment (production) & module (puppet-archive).

https://github.com/wazuh/wazuh-puppet/blob/master/manifests/indexer.pp#L67-L75

The path below works on puppet agent as well as the master 
```
path => '/etc/puppetlabs/code/environments/production/modules/archive/files/', 
```

**Comment by Hedius** [comment]
Guess community input is not wanted here.... (Same with the puppet 7 EOL issue etc.)

**Comment by nward** [comment]
Yeah, this Puppet module is pretty poor. It's very much "a customer told us they needed Puppet support so we threw it together", it's full of poor practices.

It (and the documentation) is written assuming that it is installed in a vacuum i.e. that it is the only thing installed with Puppet, which is almost never the case. If someone is wanting to use Puppet it's because they have an existing Puppet environment, and modules need to follow standard practices and work within that.
